src/adapter/src/coord/caught_up.rs

```diff
@@ -299,23 +303,44 @@ impl Coordinator {
             let within_lag =
                 PartialOrder::less_equal(live_write_frontier, &bumped_write_plus_allowed_lag);
 
-            if within_lag {
+            // This call is on the expensive side, because we have to access
+            // shared instance state, behind a lock. But it's okay because we
```

Comment:
This is not quite accurate: It's true that the call is expensive but that's not because it accesses shared state but because it performs a call to the compute controller, which is sequenced with any other work the compute controller may have to do.

Reply:
tyty! fixed up the comment